app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,118 @@
+    val homeScreenMapper: HomeScreenMapper,
+) : ViewModel() {
+
+    private val _popularMoviesHomeViewState = MutableStateFlow(HomeMovieCategoryViewState())
+    val popularMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _popularMoviesHomeViewState.asStateFlow()
+
+    private val _nowPlayingMoviesHomeViewState = MutableStateFlow(HomeMovieCategoryViewState())
+    val nowPlayingMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _nowPlayingMoviesHomeViewState.asStateFlow()
+
+    private val _upcomingMovieHomeViewState = MutableStateFlow(HomeMovieCategoryViewState())
+    val upcomingMovieHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _upcomingMovieHomeViewState.asStateFlow()
+
+    init {
+        getPopular(homeScreenMapper)
+        getUpcoming(homeScreenMapper)
+        getNowPlaying(homeScreenMapper)
+    }
+
+    fun getPopular(homeScreenMapper: HomeScreenMapper) {
+        viewModelScope.launch {
+            movieRepository.popularMovies(MovieCategory.POPULAR_STREAMING).collect {
+                _popularMoviesHomeViewState.value = homeScreenMapper.toHomeMovieCategoryViewState(
+                    movieCategories = popularCategory,
+                    selectedMovieCategory = MovieCategory.POPULAR_STREAMING,
+                    movies = it
+                )
+            }
+        }
+    }
+
+    fun getNowPlaying(homeScreenMapper: HomeScreenMapper) {
+        viewModelScope.launch {
+            movieRepository.nowPlayingMovies(MovieCategory.NOW_PLAYING_MOVIES).collect {
+                _nowPlayingMoviesHomeViewState.value =
+                    homeScreenMapper.toHomeMovieCategoryViewState(
+                        movieCategories = nowPlayingCategory,
+                        selectedMovieCategory = MovieCategory.NOW_PLAYING_MOVIES,
+                        movies = it
+                    )
+            }
+        }
+    }
+
+    fun getUpcoming(homeScreenMapper: HomeScreenMapper) {
+        viewModelScope.launch {
+            movieRepository.upcomingMovies(MovieCategory.UPCOMING_TODAY).collect {
+                _upcomingMovieHomeViewState.value = homeScreenMapper.toHomeMovieCategoryViewState(
+                    movieCategories = upcomingCategory,
+                    selectedMovieCategory = MovieCategory.UPCOMING_TODAY,
+                    movies = it
+                )
+            }
+        }
+    }
+
+
+    fun togglePopular(labelId : Int){
+        viewModelScope.launch {
+            movieRepository.popularMovies(MovieCategory.POPULAR_STREAMING).collect {
+                _popularMoviesHomeViewState.value = homeScreenMapper.toHomeMovieCategoryViewState(
+                    movieCategories = popularCategory,
+                    selectedMovieCategory = MovieCategory.getByOrdinal(labelId)!!,
+                    movies = it
+                )
+            }
+        }
+    }
+
+    fun toggleNowPlaying(labelId : Int){
```

Comment:
you're not toggling movies, you're fetching a new list - toggle means that some value is changed from 0 to 1 and vice versa. Please fix naming